index.js

```diff
@@ -0,0 +1,148 @@
+window.onload = (event) => {
+  var game = new Game();
+  document.getElementById('score-zero').onclick = () => {
+    updateFrame(0);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-one').onclick = () => {
+    updateFrame(1);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-two').onclick = () => {
+    updateFrame(2);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-three').onclick = () => {
+    updateFrame(3);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-four').onclick = () => {
+    updateFrame(4);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-five').onclick = () => {
+    updateFrame(5);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-six').onclick = () => {
+    updateFrame(6);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-seven').onclick = () => {
+    updateFrame(7);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-eight').onclick = () => {
+    updateFrame(8);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-nine').onclick = () => {
+    updateFrame(9);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-ten').onclick = () => {
+    updateFrame(10);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('reset').onclick = () => {
+    window.location.reload();
+  }
+
+  
```

Comment:
If you take a step back, maybe a good refactor could be to have two main methods. One that updates the whole interface based on the game, and one that is attached to the number buttons and enters a roll.
The rest of the logic, in-between, should entirely be in the game.
So the interface does not need to know if something is a spare or a strike, etc, the game can figure that out.

With this plan, what you need to know is:
- how can I tell the game what I've rolled. Maybe an `inputRoll` method that takes a number
- what does the game need to return so that I can fill in the interface? Probably the pins + scores for each frame. So maybe you can have the game have a method named `dataToDisplay` that could return something like `[{ "roll1" : 9,  "roll2": 1, "score":  12}, { "roll1" : 1,  "roll2": 1, "score":  14}]` ... That would be calculated on the game class based on the inputs.

Then the interface only job is to link the button click to the right inputRoll, and then use the data to display to add the correct text to the HTML.

Let me know if that makes sense.


Then you can develop the game class, using tests that s